### PR TITLE
Crear Inbound en Almex

### PR DESCRIPTION
**Pull request by afska**
[Finishes #99611294](https://www.pivotaltracker.com/story/show/99611294)

Uso:

``` coffee-script
AlmexApi = require("almex-sdk")

almexApi = new AlmexApi
  username: "**********"
  password: "********"
  accountId: "1241"

inbound =
  id: 9825
  date: new Date("2015-08-25T19:38:45.237Z")
  products: [
    id: "232"
    quantity: 5
    description: "Camiseta de River Plate"
    barcode: "B0CACACA"
  ,
    id: "233"
    quantity: 1
    description: "Estampilla"
    barcode: "0CA0CA"
  ]

almexApi.createInputBean inbound
```

### CODE REVIEW

**Comment by andreskir** [comment]
:+1:

File: src/almexInboundsAdapter.spec.coffee [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        id: 9
+        fecha: "25/08/2015"
+        total: 2
+        cantidadTotal: 6
+        listProducto: [
+          sku: "232"
+          descripcion: "Camiseta de River Plate"
+          cantidad: 5
+          idWb: 1
+          ean: "B0CACACA"
```
Comment (Gustrucco) [inline src/almexInboundsAdapter.spec.coffee:35]: @andreskir viste esto?

**Comment by Gustrucco** [comment]
:wine_glass:

File: src/almexApi.coffee [hunk 2]
```diff
@@ -47,31 +48,65 @@ class AlmexApi
...
+
+    req.postAsync(params).spread (response) =>
+      xml2js.parseStringAsync response.body
 
   _getResult: (xml, method) =>
     result = try xml["S:Envelope"]["S:Body"][0]["ns2:#{method}Response"][0].return
     if not result?
       # this should never happen
-      throw new Error "Los web services de la vida, no son lo que yo esperaba, no son lo que yo creía, no son lo que imaginaba..."
+      throw new Error "Los web services de la vida, no son lo que yo esperaba, no son lo que yo creía, no son lo que imaginaba...\n" + JSON.stringify xml
```
Comment (NahuelP) [inline src/almexApi.coffee:112]: Jajjajaja

File: build/almexInboundsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    function AlmexInboundsAdapter() {
+      this.getInputBean = __bind(this.getInputBean, this);
+    }
+
+    AlmexInboundsAdapter.prototype.getInputBean = function(inbound) {
+      var date;
+      date = inbound.date.toISOString();
+      return {
+        id: inbound.id,
+        fecha: date.substr(0, date.indexOf("T")).split("-").reverse().join("/"),
```
Comment (NahuelP) [inline build/almexInboundsAdapter.js]: (1): [momentjs](http://momentjs.com/)

File: src/almexInboundsAdapter.coffee [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+# Converts inbounds to Almex's input beans.
+class AlmexInboundsAdapter
+  getInputBean: (inbound) =>
+    date = inbound.date.toISOString()
+
+    id: inbound.id
+    fecha: date
+      .substr(0, date.indexOf("T"))
+      .split("-").reverse().join "/"
```
Comment (NahuelP) [inline src/almexInboundsAdapter.coffee]: (1)

**Comment by NahuelP** [comment]
:cocktail:

**Comment by charlyraffellini** [comment]
:kiss:

**Pull request merged**